financialaid/views.py

```diff
@@ -1,24 +1,229 @@
+                    sort_direction=new_sort_direction if field == self.sort_field else "",
+                    sort_field=field
+                ),
+                # If this field is the current sort field, we want to indicate the current sort direction
+                "direction_display": self.sort_direction if field == self.sort_field else None,
+                "display": self.sort_fields[field]["display"]
+            }
+            for field in sort_link_order
+        )
+
+        # Required for styling
+        context["style_src"] = get_bundle_url(self.request, "style.js")
+        context["dashboard_src"] = get_bundle_url(self.request, "dashboard.js")
+        js_settings = {
+            "gaTrackingID": settings.GA_TRACKING_ID,
+            "reactGaDebug": settings.REACT_GA_DEBUG,
+            "authenticated": not self.request.user.is_anonymous(),
+            "edx_base_url": settings.EDXORG_BASE_URL,
+        }
+        context["js_settings_json"] = json.dumps(js_settings)
+        return context
+
+    def get_queryset(self):
+        """
+        Gets queryset for ListView to return to view
+        """
+        # Filter by availability
+        financial_aids = FinancialAid.objects.filter(
+            tier_program__program__live=True,
+            tier_program__program__financial_aid_availability=True
+        )
+
+        # Filter by program (self.program set in test_func())
+        financial_aids = financial_aids.filter(
+            tier_program__program=self.program,
+        )
+
+        # Filter by status
+        self.selected_status = self.kwargs.get("status", None)
+        if self.selected_status is None or self.selected_status not in FinancialAidStatus.ALL_STATUSES:
+            self.selected_status = FinancialAidStatus.PENDING_MANUAL_APPROVAL
+        financial_aids = financial_aids.filter(status=self.selected_status)
+
+        # Sort by field
+        self.sort_field = self.request.GET.get("sort_by", self.default_sort_field)
+        if self.sort_field.startswith("-"):
+            self.sort_field = self.sort_field[1:]
+            # Defined above: self.sort_direction = ""
+            self.sort_direction = "-"
+        if self.sort_field not in self.sort_fields:
+            self.sort_field = self.default_sort_field
+            self.sort_direction = ""
+        financial_aids = financial_aids.order_by(
+            "{sort_direction}{sort_field}".format(
+                sort_direction=self.sort_direction,
+                sort_field=self.sort_field_mappings.get(self.sort_field, self.sort_field)
+            )
+        )
+
+        # Add adjusted cost and TierPrograms valid for selection for each FinancialAid object
+        # Note: iterating through this queryset, if large, can be very costly in terms of memory/time,
+        # but it is done here to simplify the templates.
+        # Get program ids and associated tier programs
+        program_ids = Program.objects.filter(
```

Comment:
Why are we making this list here? `self.program.id` is the only id which matters here